packages/golden_toolkit/lib/src/testing_tools.dart

```diff
@@ -199,7 +199,7 @@ Future<void> screenMatchesGolden(
     finder: finder,
     customPump: customPump,
     skip: skip,
-    device: null,
+    device: Device.phone,
```

Comment:
I'm not sure this is the correct behavior to maintain backwards compatibility.

Some goldens are device agnostic. I believe the DeviceFileNameFactory optionally takes a device so that implementors can incorporate it into their file naming. However, it is not always present. I think this should remain null.

Reply:
Nevermind... I studied the code more. It is structured very poorly. In this code path, this device parameter is completely ignored (because right now, it is ultimately passed in the anonymous func immediately below this line.

I think this should be refactored, because it feels like a trap that will someday bite us. However, you don't need to fix it. We just need to do our own house cleaning 😄